src/recipe/Baked.Recipe.Service.Application/CodingStyle/RichTransient/InitializeUsingIdParameterConvention.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Baked.Business;
+using Baked.Domain.Model;
+using Baked.RestApi.Configuration;
+using Baked.RestApi.Model;
+using Humanizer;
+
+namespace Baked.CodingStyle.RichTransient;
+
+public class InitializeUsingIdParameterConvention(DomainModel _domain)
+    : IApiModelConvention<ActionModelContext>
+{
+    public void Apply(ActionModelContext context)
+    {
+        if (!context.Controller.MappedType.TryGetMembers(out var members)) { return; }
+        if (!members.Methods.Having<InitializerAttribute>().Any()) { return; }
+        if (!members.Has<LocatableAttribute>()) { return; }
+        if (context.Controller.MappedType.TryGetQueryType(_domain, out var _)) { return; }
+        if (context.Action.MappedMethod is null) { return; }
+        if (context.Action.MappedMethod.Has<InitializerAttribute>()) { return; }
+
+        var initializer = members.Methods.Having<InitializerAttribute>().Single();
+        if (!initializer.DefaultOverload.Parameters.TryGetValue("id", out var parameter)) { return; }
+
+        context.Action.Parameter["id"] =
+            new(parameter.ParameterType, ParameterModelFrom.Route, parameter.Name, MappedParameter: parameter)
+            {
+                IsOptional = parameter.IsOptional,
+                DefaultValue = parameter.DefaultValue,
+                IsInvokeMethodParameter = false,
+                RoutePosition = 1,
+                AdditionalAttributes = [$"SwaggerSchema(\"Unique value to initialize {context.Controller.MappedType.Name.Humanize().ToLowerInvariant()} resource\")"]
```

Comment:
don't say `initialize`, keep the same wording with entity. rich transients may still "find" a resource with given id. this difference is irrelevant to api users.